src/bin/test_providers.rs

```diff
@@ -425,24 +423,39 @@ impl Providers {
 }
 
 async fn get_latest_block_number() -> Result<u64> {
-    let config = Config::new()
-        .add_header("Content-Type", "application/json")
-        .unwrap()
-        .add_header("CF-Access-Client-Id", PANDAOPS_CLIENT_ID.to_string())
-        .unwrap()
-        .add_header(
-            "CF-Access-Client-Secret",
-            PANDAOPS_CLIENT_SECRET.to_string(),
-        )
-        .unwrap()
-        .set_base_url(Url::parse("https://geth-lighthouse.mainnet.eu1.ethpandaops.io/").unwrap());
-    let client: Client = config.try_into()?;
-    let params = Params::Array(vec![json!("latest"), json!(false)]);
-    let method = "eth_getBlockByNumber".to_string();
-    let request = JsonRequest::new(method, params, 1);
-    let response = client.post("").body_json(&request).unwrap();
-    let response = response.recv_string().await.unwrap();
-    let response = serde_json::from_str::<serde_json::Value>(&response)?;
+    let mut headers = HeaderMap::new();
+    headers.insert(CONTENT_TYPE, HeaderValue::from_static("application/json"));
+    headers.insert(
+        "CF-Access-Client-Id",
+        HeaderValue::from_str(&PANDAOPS_CLIENT_ID)
+            .map_err(|_| anyhow!("Invalid CF-Access-Client-Id header value"))?,
+    );
+    headers.insert(
+        "CF-Access-Client-Secret",
+        HeaderValue::from_str(&PANDAOPS_CLIENT_SECRET)
+            .map_err(|_| anyhow!("Invalid CF-Access-Client-Secret header value"))?,
+    );
+    let client = Client::builder()
```

Comment:
nit: I think this can be simplified to something like:

```rust
let response = Client::new()
    .post("...")
    .headers(headers)
    .json(&request)
    .send()
    .await
    .map_err(|e| ...)?
```